src/test/java/com/spotify/folsom/KetamaIntegrationTest.java

```diff
@@ -70,6 +72,22 @@ public static void tearDownClass() throws Exception {
     binaryServers.stop();
   }
 
+  public static void allClientsConnected(final MemcacheClient<?> client)
+      throws Exception{
+    final CountDownLatch done = new CountDownLatch(1);
+
+    client.registerForConnectionChanges(new ConnectionChangeListener() {
```

Comment:
Well, I opted for the simpler solution here. In the future I imagine that we will provide utility functions to have a future return when say 90% of a cluster is up, but I think that we should put some more thinking into the semantics of that.

With countdown latch it is easy to have it timeout as well (even though I managed to misremember the fact that it doesn't throw an exception when timing out but instead just returns false. I'll fix that in an upcoming patch)